test_lib/sla.py

```diff
@@ -217,9 +218,13 @@ def attached_service_level(self):
     def attached_service_level_name(self):
         return self._attached_service_level.name
 
-    def attach_service_level(self, service_level):
+    # By Eliran:
+    # If a cluster is greater than 3 nodes, the information might not have been propagated to the node that
+    # does the attachment. We have to wait until all nodes have added the service level
+    @retrying(n=6, sleep_time=5, message="Waiting until all nodes have added the service level")
```

Comment:
Thanks for making the change, I would have done this differently but I might missed something. Putting the retrying here
will force this call to last 30 seconds when it is supposed to fail.

Maybe it is better to call this locally or add a new function - retrying_attach_service_level which does it. (?)
But if you think it is never going to be used that way so it is not necesairy.

An example:
after your change, if I add a line like the below or something similar the assert will either fail very fast
of it it passes it will take 30 seconds to continue the test.
`assert not role.attach_service_level(some_service)`